### PR TITLE
HW03: Arthur Bohan

### PR DESCRIPTION
**Pull request by arthurbohan**

### CODE REVIEW

File: HW03/Arthur_Bohan/student.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# student
+class Student
+  attr_reader :name, :surname
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+    @student_homeworks = []
```
Comment (violettaMi) [inline HW03/Arthur_Bohan/student.rb:10]: you can use just '@homeworks' cos it's already in the Student class.

File: HW03/Arthur_Bohan/student.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @mentors = []
+  end
+
+  def submit_homework!(homework)
+    submitted_homework = Homework.new(homework)
+    @student_homeworks << submitted_homework
```
Comment (violettaMi) [inline HW03/Arthur_Bohan/student.rb:16]: please use 'attr_reader' for 'homeworks'

File: HW03/Arthur_Bohan/student.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    notification(homework_notification(homework))
+  end
+
+  def homeworks
+    p @student_homeworks
+  end
```
Comment (violettaMi) [inline HW03/Arthur_Bohan/student.rb:22]: this method will be unnecessary if you use 'attr_reader'

File: HW03/Arthur_Bohan/notification.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+# notification
+class Notification
+  attr_reader :notification_data
+
+  def initialize(_notification_data)
+    @notification_data
+  end
```
Comment (violettaMi) [inline HW03/Arthur_Bohan/notification.rb:9]: mb do you mean smth like this?
```suggestion
  def initialize(notification_data)
    @notification_data = notification_data
  end
```

File: HW03/Arthur_Bohan/mentor.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+# mentor
+class Mentor
+  attr_reader :name, :surname
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+    @notifications = []
+    @students = []
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/mentor.rb:11]: Why do you need this variable if you don't use it? 🤔

File: HW03/Arthur_Bohan/student.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+# student
+class Student
+  attr_reader :name, :surname, :homeworks
+
+  def initialize(name:, surname:)
+    @name = name
+    @surname = surname
+    @homeworks = []
+    @mentors = []
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/student.rb:11]: Why do you need this variable if you don't use it? 🤔

File: HW03/Arthur_Bohan/mentor.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def subscribe_to(student)
+    student.submitted_mentor(self)
+  end
+
+  def notification(message)
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/mentor.rb:18]: Please, use a more descriptive name, if you add notification use `add_notification` name

File: HW03/Arthur_Bohan/mentor.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    notification = Notification.new(message)
+    @notifications << notification
+  end
+
+  def notifications
+    p @notifications
+  end
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/mentor.rb:25]: If you want to show notifications use `show_notifications` name, in another case, you can just add `attr_reader` for `notifications`

File: HW03/Arthur_Bohan/student.rb [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    submitted_homework = Homework.new(homework)
+    @homeworks << submitted_homework
+    notification(homework_notification(homework))
+  end
+
+  def homework_notification(homework)
+    puts "#{name} #{surname} submit: #{homework}"
+  end
+
+  def notification(request)
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/student.rb:24]: Please, use `verbs` for method names if they perform any action and `nouns` if they just return a value.

For example
```rb
    def show_homework_notification(homework)
      puts "#{name} #{surname} submit: #{homework}"
    end

    def homework_notification_message(homework)
       "#{name} #{surname} submit: #{homework}"
    end
```

File: HW03/Arthur_Bohan/student.rb [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    @mentor&.notification(request)
+  end
+
+  def submitted_mentor(current_mentor)
+    @mentor = current_mentor
```
Comment (vladhilko) [inline HW03/Arthur_Bohan/student.rb:29]: Is it possible to add many `mentors` to one `student`? 🤔 Could you try to implement it?

**Comment by vladhilko** [comment]
The solution was updated and rewritten.

**Pull request merged**